Task4/src/main/java/by/innowise/tasks/mapper/StudyHourMapper.java

```diff
@@ -0,0 +1,42 @@
+package by.innowise.tasks.mapper;
+
+import by.innowise.tasks.dto.EventDto;
+import by.innowise.tasks.dto.LessonDto;
+import by.innowise.tasks.dto.StudyHourDto;
+import by.innowise.tasks.entity.Event;
+import by.innowise.tasks.entity.Lesson;
+import by.innowise.tasks.entity.StudyHour;
+import by.innowise.tasks.handler.UnknownClassException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class StudyHourMapper {
+
+    @Autowired
+    private EventMapper eventMapper;
+
+    @Autowired
+    private LessonMapper lessonMapper;
+
+    public StudyHourDto toStudyHourDto(StudyHour studyHour) {
+
+        if (studyHour instanceof Event) {
+            return eventMapper.toEventDto((Event) studyHour);
```

Comment:
We can avoid such stupid logic but I have to show you how ^__^. Remind me when you'll be in office.